### DISCUSSION TITLE
Is it possible to use InertiaJS without any middleware?

### DISCUSSION THREAD

**Discussion by blackjyn** [inertiajs/inertia#1579] [Help]
Sorry for a may be silly question: Is it possible to use InertiaJS without any middleware? 
Just pure native PHP code?
Thanks!

**Comment by aadrian** [comment]
@blackjyn 

> Is it possible to use InertiaJS without any middleware? Just pure native PHP code?

Sure it is, but than you need to implement the `adapter` yourself in "pure native PHP" !  

The community already implemented a few: for https://inertiajs.com/community-adapters 

There are even more https://github.com/inertiajs/inertiajs.com/pulls?q=is%3Apr+is%3Aopen+adapter waiting (some since more than a year :( ) to be merged.

**Reply by blackjyn** [reply]
Sorry. That's means we still require adapter right? 
Looks everything is tightly coupled with server side framework/library. 

What I expected is a loosely coupled approach without the use of specific server side  library/framework.

I can't even find a simple Hello World sample on it. inertiajs promotes Laravel from the beginning.

What I expected is by "pure native PHP" is just minimum overhead without any specific server side library/framework 

Really looking into it.

**Reply by aadrian** [reply]
@blackjyn 
>  That's means we still require adapter right?

Yes. The community made many of such adapters as I linked above.

> Looks everything is tightly coupled with server side framework/library.

Please take a look at the documentation. 
For Inertia to be able to do what it does, it needs the data from the server in a special format. This is what the "adapter" needs to provide (with a framework/library or without it).

> I can't even find a simple Hello World sample on it. 

There's no "simple Hello World". It's quite "boolean": one does implement the communication protocol or not https://inertiajs.com/the-protocol


> inertiajs promotes Laravel from the beginning.

No, they don't "promote" it. 
They just did a **reference implementation** that they keep up to date, so that it's easier for the community to replicate that for the different frameworks or language stacks.
